File: test_connection.py
#!/usr/bin/env python3
"""Test script to verify App Store Connect API connection."""

import os
from dotenv import load_dotenv
from python.app_store_connect import Client

# Load environment variables
load_dotenv()

def test_connection():
    """Test the App Store Connect API connection."""
    try:
        # Initialize the client
        client = Client(
            key_id=os.getenv("ASC_KEY_ID"),
            issuer_id=os.getenv("ASC_ISSUER_ID"),
            private_key_path=os.getenv("ASC_PRIVATE_KEY_PATH")
        )
        
        print("✅ Client initialized successfully")
        
        # Try to fetch apps (this will validate the authentication)
        apps = client.apps.get_all()
        print(f"✅ Successfully connected! Found {len(apps)} apps")
        
        # Show app details if any exist
        if apps:
            for app in apps[:3]:  # Show first 3 apps
                print(f"  - {app.get('attributes', {}).get('name', 'Unknown')} ({app.get('attributes', {}).get('bundleId', 'Unknown')})")
        
        return True
        
    except Exception as e:
        print(f"❌ Connection failed: {e}")
        return False

if __name__ == "__main__":
    test_connection()

File: rust/examples/test_connection.rs
use app_store_connect_rust::{Client, AppStoreConnectError, AuthenticationError};
use std::env;

#[tokio::main]
async fn main() -> Result<(), AppStoreConnectError> {
    // Load environment variables from parent directory's .env
    let _ = dotenvy::from_path("../.env");
    
    // Get credentials
    let key_id = env::var("ASC_KEY_ID")
        .map_err(|_| AppStoreConnectError::Authentication(AuthenticationError::new("ASC_KEY_ID not set")))?;
    let issuer_id = env::var("ASC_ISSUER_ID")
        .map_err(|_| AppStoreConnectError::Authentication(AuthenticationError::new("ASC_ISSUER_ID not set")))?;
    let mut private_key_path = env::var("ASC_PRIVATE_KEY_PATH")
        .map_err(|_| AppStoreConnectError::Authentication(AuthenticationError::new("ASC_PRIVATE_KEY_PATH not set")))?;
    
    // If path starts with ./, make it relative to parent directory
    if private_key_path.starts_with("./") {
        private_key_path = format!("../{}", private_key_path.trim_start_matches("./"));
    }
    
    // Initialize client
    let client = Client::new(&key_id, &issuer_id, &private_key_path).await?;
    println!("✅ Client initialized successfully");
    
    // Try to fetch apps
    match client.apps().get_all(Some(10)).await {
        Ok(apps) => {
            println!("✅ Successfully connected! Found {} apps", apps.len());
            
            // Show first 3 apps
            for app in apps.iter().take(3) {
                if let Some(attrs) = app.get("attributes") {
                    let name = attrs.get("name")
                        .and_then(|n| n.as_str())
                        .unwrap_or("Unknown");
                    let bundle_id = attrs.get("bundleId")
                        .and_then(|b| b.as_str())
                        .unwrap_or("Unknown");
                    println!("  - {} ({})", name, bundle_id);
                }
            }
        }
        Err(e) => {
            println!("❌ Connection failed: {}", e);
        }
    }
    
    Ok(())
}